distribution/scripting.md

```diff
@@ -69,40 +131,83 @@ JavaScript is a very mature and flexible language with a large, if not the large
+Scripts can consist of any behaviour and have a large memory pool available to them. The speed will vary depending on the hardware and system executing them. The scripts are interpreted, so do not expect anywhere close to the performance of native code. In most scenarios this should be satisfactory, but a random map generator, or genetic algorithm for building roller coasters might struggle. Like any language, there will be tricks to optimising JavaScript and the use of the OpenRCT2 APIs. [Duktape also provides some engine-specific performance tips](https://wiki.duktape.org/performance).
+
+The APIs for OpenRCT2 try to provide access to the game’s data structures as much as possible but we can only add so many at a time. The best way to grow the plugin system is to add APIs on-demand. So if you find an API is missing, please raise an issue for it on GitHub and also feel free to discuss it on our Discord and to submit a pull request afterwards.
+
+---
 
-The APIs for OpenRCT2 try to mimic the internal data structures as close as possible but we can only add so many at a time. The best way to grow the plug-in system is to add APIs on-demand. So if you find an API is missing, please raise an issue for it on GitHub and also feel free to submit a pull request afterwards.
+> What is `targetApiVersion`?
 
-> What is ```targetApiVersion```
+In case there are breaking API changes, plugins can use this to keep the old API behaviour. For example in version 34 `Entity.type` would no longer return `peep` for guests and staff, instead it would return either `guest` or `staff`, so if your plugin expects `peep` you can specify the version 33 to keep the old behaviour. See the [list of breaking changes](#breaking-changes). If this is not specified it will default to version 33 and the plugin will log an error on startup, it is recommended to specify the current API version.
 
-In case there are breaking Api changes plugins can use this to keep the old Api behavior, as an example in version 34 ```Entity.Type``` would no longer return "peep" for
-guests and staff, instead it would return either "guest" or "staff", so if your plugin expects "peep" you can specify the version 33 to keep the old behavior. See the list
-of breaking changes. If this is not specified it will default to version 33, it is recommended to specify the current api version.
+---
+
+> How do I prevent my script from running on older versions of OpenRCT2 that do not support all the APIs I require?
+
+When registering your plugin, you can specify a minimum required API version like below:
+
+```js
+registerPlugin({
+    name: 'Your Plugin',
+    version: '1.2', // Your plugin version
+    authors: ['Your Name'],
+    type: 'remote',
+    minApiVersion: 7, // OpenRCT2 plugin API v7 or higher is required to run your plugin
+    main: main
```

Comment:
Plugins should also specify their licence. (We generally recommend `MIT` as the default licence here.)